include/GafferScene/Private/IECoreScenePreview/Renderer.h

```diff
@@ -183,6 +188,14 @@ class IECORESCENE_API Renderer : public IECore::RefCounted
 				/// such as Arnold where the attributes are not orthogonal to the
 				/// geometric representation.
 				virtual bool attributes( const AttributesInterface *attributes ) = 0;
+				/// Declares links between this object and others. Standard link
+				/// types are :
+				///
+				/// - "lights" : specifies the set of lights that should illuminate
+				///   an object.
+				/// - "lightFilters" : specifies the set of light filters that should
+				///   be applied to a light.
+				virtual void link( const IECore::InternedString &type, const ConstObjectSetPtr &objects ) = 0;
```

Comment:
The immediate reason is Arnold's `shadow_group`, which we want to support, but don't consider to be a standard/universal thing. I'm also hedging my bets a bit in case other types of renderer-specific linking come up.